src/nsj_rest_lib/service/service_base.py

```diff
@@ -894,10 +898,13 @@ def _save(
 
             old_dto = None
             # Recuperando o DTO antes da gravação (apenas se for update, e houver um custom_after_update)
-            if not insert:
+            if not insert and not upsert:
                 old_dto = self._retrieve_old_dto(dto, id, aditional_filters)
                 setattr(dto, dto.pk_field, getattr(old_dto, dto.pk_field))
 
+            if not insert and upsert:
```

Comment:
O mesmo de cima. Mas, memso que não usemos no caso de update, acho que teria faltado um "not" aqui...